processor/websocketprocessor/processor.go

```diff
@@ -0,0 +1,175 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package websocketprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/websocketprocessor"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/obsreport"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/collector/processor"
+	"go.uber.org/zap"
+	"golang.org/x/net/websocket"
+)
+
+type wsprocessor struct {
+	config            *Config
+	telemetrySettings component.TelemetrySettings
+	obsproc           *obsreport.Processor
+	logsSink          consumer.Logs
+	metricsSink       consumer.Metrics
+	tracesSink        consumer.Traces
+	server            *http.Server
+	shutdownWG        sync.WaitGroup
+	connections       map[string]chan []byte
+	connLock          sync.RWMutex
+}
+
+var processors = map[*Config]*wsprocessor{}
+
+func newProcessor(settings processor.CreateSettings, config *Config) (*wsprocessor, error) {
+	if p, ok := processors[config]; ok {
+		return p, nil
+	}
+	obsproc, err := obsreport.NewProcessor(obsreport.ProcessorSettings{
+		ProcessorID:             settings.ID,
+		ProcessorCreateSettings: settings,
+	})
+	if err != nil {
+		return nil, err
+	}
+	conns := make(map[string]chan []byte)
+	p := &wsprocessor{
+		config:            config,
+		obsproc:           obsproc,
+		telemetrySettings: settings.TelemetrySettings,
+		connections:       conns,
+	}
+	processors[config] = p
```

Comment:
Is this the same as `internal/sharedcomponent`?

Reply:
yep, sharedcomponent doesn't implement the method Capabilities.